tests/lib/rules/prefer-stateless-function.js

```diff
@@ -496,6 +496,39 @@ ruleTester.run('prefer-stateless-function', rule, {
           }
         }
       `,
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      code: `
+        class Foo extends React.Component {
+          static defaultProps = {
+            foo: boolean
+          }
+          render() {
+            const { foo } = this.props;
+            return foo ? <div /> : null;
+          }
+        }
+      `,
+      parser: 'babel-eslint',
+      errors: [{
+        message: 'Component should be written as a pure function'
+      }]
+    }, {
+      code: `
+        class Foo extends React.Component {
+          static get defaultProps() {
```

Comment:
Let's also add a test case with `Foo.defaultProps = {}`, that doesn't use the babel-eslint parser (also this test case shouldn't need it either)